### PR TITLE
Ae macosx fix

### PR DESCRIPTION
**Pull request by AEscobarCruz**
Original fixes included checking for hidden apple files when compressing folders. Additional fixes include a bit more logging, and some refactoring to FeatureValidator to make it more functional and easier to read.

### CODE REVIEW

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala [hunk 1]
```diff
@@ -62,24 +52,57 @@ object FeatureValidator extends Logging {
...
+  }
+
+  /**
+   * Compare geometry complexity with max in configs.
+   * @param complexity
+   * @param maxComplexity
+   * @return
+   */
+  private def checkComplexity(complexity: Int, maxComplexity: Int): Result = {
+    if(complexity > maxComplexity)
```
Comment (velvia) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: space after if

**Comment by velvia** [comment]
Reviewed up to fa022e5

File: geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceServlet.scala [hunk 2]
```diff
@@ -71,7 +71,10 @@ class GeospaceServlet(sodaFountain: SodaFountainClient,
             logger.info("Feature validation bypassed")
           } else {
             val validationErrors = FeatureValidator.validationErrors(features, myConfig.maxMultiPolygonComplexity)
-            if (!validationErrors.isEmpty) halt(BadRequest(validationErrors))
+            if (!validationErrors.isEmpty){
+              logger.warn("Feature validation failed! Fail count {}, first failure: {} ...aborting!", validationErrors.size, validationErrors.head.toString)
```
Comment (sokrahta) [inline geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceServlet.scala]: Style: this line is too long over 120 chars. Try splitting over multiple lines.

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala [hunk 3]
```diff
@@ -62,24 +52,57 @@ object FeatureValidator extends Logging {
...
+
+  /**
+   * Validate multipolygon object
+   * @param mp
+   * @param maxMultiPolygonComplexity
+   * @return
+   */
+  private def validateMultiPolygon(mp: MultiPolygon, maxMultiPolygonComplexity: Int): Result = {
+    if (!mp.isValid)
+      return GeometryNotValid
```
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Style: return is discouraged in scala, try rearranging the conditional
```
if (mp.isValid) {
   ... do some stuff ...
} else {
  GeometryNotValid
}
```

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala [hunk 4]
```diff
@@ -62,24 +52,57 @@ object FeatureValidator extends Logging {
...
+  //  -90 <= lat <= 90
+  // If we switch projections, this logic will need to be updated
+  // The values below are slightly relaxed since reprojection can result in rounding to values like 180.00000041
+  // We only care about precision to 6 decimal places so this should be acceptable.
+  private def offTheMapPoints(coordinates: Array[Coordinate]): Result ={
+    val badCoordinates = coordinates.filter { c =>
+      c.x >= 180.000001 || c.x <= -180.000001 || c.y >= 90.000001 || c.y <= -90.000001
+    }
+    if (!badCoordinates.isEmpty)
+      return GeometryContainsOffMapPoints(badCoordinates)
```
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Style: avoid using return, try like this instead
```
if (badCoordinates.isEmpty) Valid else GeometryContainsOffMapPoints(badCoordinates)
```

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala (hunk 1 above)
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Additionally style: avoid using return. 
```
if (complexity > maxComplexity) GeometryTooComplex(complexity, maxComplexity) else Valid
```

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala (hunk 3 above)
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Also make sure to use braces when `if/else` crosses multiple lines.

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala (hunk 4 above)
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Also make sure to use braces when `if/else` crosses multiple lines.

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala (hunk 1 above)
Comment (sokrahta) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/feature/FeatureValidator.scala]: Also make sure to use braces when `if/else` crosses multiple lines.

**Comment by AEscobarCruz** [comment]
To be addressed later.

**Pull request closed without merging**